openmc/mgxs/library.py

```diff
@@ -1092,60 +1155,121 @@ def create_mg_mode(self, xsdata_names=None, xs_ids=None,
 
         if xsdata_names is not None:
             cv.check_iterable_type('xsdata_names', xsdata_names, basestring)
+        xs_type = 'macro'
+
+        # Initialize MGXS File
+        mgxs_file = openmc.MGXSLibrary(self.energy_groups)
+
+        # Set ID names
         if xs_ids is not None:
             if isinstance(xs_ids, basestring):
                 # If we only have a string lets convert it now to a list
                 # of strings.
-                xs_ids = [xs_ids for i in range(len(self.domains))]
+                xs_id_input = xs_ids
+                xs_ids = []
+                for domain in self.domains:
+                    if self.domain_type is 'mesh':
+                        for subdomain in domain.cell_generator():
```

Comment:
And actually, I think you can simplify further within something like:

```python
all_xs_ids = []
for domain in self.domains:
    if self.domain_type is 'mesh':
        all_xs_ids.extend([xs_ids] * domain.num_mesh_cells)
    else:
        all_xs_ids.append(xs_ids)
```

I may be picky, but I typically don't like the redefinition of variables as you do with `xs_id_input` and `xs_ids` since it often is not the most succinct way to express an algorithm.